server/src/main/java/io/unitycatalog/server/persist/UserRepository.java

```diff
@@ -0,0 +1,191 @@
+package io.unitycatalog.server.persist;
+
+import io.unitycatalog.control.model.CreateUser;
+import io.unitycatalog.control.model.UpdateUser;
+import io.unitycatalog.control.model.User;
+import io.unitycatalog.server.exception.BaseException;
+import io.unitycatalog.server.exception.ErrorCode;
+import io.unitycatalog.server.persist.dao.UserDAO;
+import io.unitycatalog.server.persist.utils.HibernateUtils;
+import io.unitycatalog.server.persist.utils.PagedListingHelper;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UserRepository {
+  private static final UserRepository INSTANCE = new UserRepository();
+  private static final Logger LOGGER = LoggerFactory.getLogger(UserRepository.class);
+  private static final SessionFactory SESSION_FACTORY = HibernateUtils.getSessionFactory();
+  private static final PagedListingHelper<UserDAO> LISTING_HELPER =
+      new PagedListingHelper<>(UserDAO.class);
+
+  private UserRepository() {}
+
+  public static UserRepository getInstance() {
+    return INSTANCE;
+  }
+
+  public User createUser(CreateUser createUser) {
```

Comment:
According to line 23 in `control.yaml`, isn't the input to `createUser` a `UserResource`, which is different from `CreateUser`?